massa-models/src/amount.rs

```diff
@@ -9,32 +9,47 @@ use std::str::FromStr;
 
 pub const AMOUNT_ZERO: Amount = Amount::from_raw(0);
 
+/// A structure representing a decimal Amount of coins with safe operations
+/// this allows ensuring that there is never an uncontrolled overflow or precision loss
+/// while providig a convenient decimal interface for users
+/// The underlying u64 raw representation if a fixed-point value with factor AMOUNT_DECIMAL_FACTOR
+/// The minimal value is 0 and the maximal value is 18446744073.709551615
 #[derive(Clone, Copy, Debug, PartialEq, Eq, Ord, PartialOrd, Default)]
 pub struct Amount(u64);
 
 impl Amount {
+    /// obtains the underlying raw u64 representation
+    /// Warning: do not use this unless you know what you are doing
+    /// because the raw value does not take the AMOUNT_DECIMAL_FACTOR into account
     pub fn to_raw(&self) -> u64 {
         self.0
     }
 
+    /// constructs an Amount from the underlying raw u64 representation
+    /// Warning: do not use this unless you know what you are doing
+    /// because the raw value does not take the AMOUNT_DECIMAL_FACTOR into account
```

Comment:
Maybe add some thing along the lines of "prefer from_string whenever you can"

Reply:
done